core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java

```diff
@@ -72,11 +61,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         currentFieldName = parser.currentName();
                     } else {
                         if ("value".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.text();
```

Comment:
I changed this to parser.text() because the original builder only takes strings. This should still correctly parse `1`, only that it will be converted to `"1"`. Do we want to make all query builders that original would take strings also take objects such as number, BytesRef or dates? I am happy either way.